src/frontend/js/widgets/Dashboard/components/DashboardOrderLoader/index.tsx

```diff
@@ -18,41 +20,56 @@ const messages = defineMessages({
     description: 'Banner displayed when the contract is not signed',
     id: 'components.DashboardOrderLoader.signatureNeeded',
   },
+  wrongLinkedProductError: {
+    defaultMessage: 'This page is not available for this order.',
+    description: "Banner displayed when order's linked product type is not handle.",
+    id: 'components.DashboardOrderLoader.wrongLinkedProductError',
+  },
 });
 
 export const DashboardOrderLoader = () => {
+  const [wrongLinkedProductError, setWrongLinkedProductError] = useState<string>();
```

Comment:
No need to use a state here, you could use a `useMemo`

Reply:
i don't think that i can use a `useMemo` in `useEffect` l.45

Reply:
Or you want me to replace the useEffect like 
```ts
  const credentialOrder = order && isCredentialOrder(order) ? order : undefined;
  const wrongLinkedProductError = useMemo(() => {
    if (order && !credentialOrder) {
      handle(new Error("Order's details page only accept order linked to CREDENTIAL product."));
      return intl.formatMessage(messages.wrongLinkedProductError);
    }
  }, [credentialOrder]);
  const error = errorOrder || errorCourseProduct || wrongLinkedProductError;
```

Reply:
Yup that's my point. In this way you prevent an extra render loop